ceremonies/src/lib.rs

```diff
@@ -899,6 +899,63 @@ impl<T: Config> Pallet<T> {
 			.collect()
 	}
 
+	pub fn get_aggregated_account_data(
+		cid: CommunityIdentifier,
+		account: &T::AccountId,
+	) -> AggregatedAccountData<T::AccountId, T::Moment> {
+		let cindex = <encointer_scheduler::Pallet<T>>::current_ceremony_index();
+		let aggregated_account_data_global = AggregatedAccountDataGlobal {
+			ceremony_phase: <encointer_scheduler::Pallet<T>>::current_phase(),
+			ceremony_index: cindex,
+		};
+
+		let aggregated_account_data_personal: Option<
+			AggregatedAccountDataPersonal<T::AccountId, T::Moment>,
+		>;
+
+		// check if the participant is registered. if not the entire personal field will be None.
+		if let Some(participant_type) = Self::get_participant_type((cid, cindex), account) {
+			let mut meetup_location_index: Option<MeetupIndexType> = None;
+			let mut meetup_time: Option<T::Moment> = None;
+			let mut meetup_registry: Option<Vec<T::AccountId>> = None;
+			let mut meetup_index: Option<MeetupIndexType> = None;
+
+			// check if the participant is already assigned to a meetup
+			if let Some(participant_meetup_index) = Self::get_meetup_index((cid, cindex), account) {
+				meetup_index = Some(participant_meetup_index);
+				let locations = <encointer_communities::Pallet<T>>::get_locations(&cid);
+				let location_assignment_params = Self::assignments((cid, cindex)).locations;
+
+				meetup_location_index = get_meetup_location_index(
+					participant_meetup_index,
+					&locations,
+					location_assignment_params,
+				);
+				let location =
+					Self::get_meetup_location((cid, cindex), participant_meetup_index).unwrap();
```

Comment:
No unwraps in runtime code. If there is a formal proof that this does not panic, use `expect("proof; qed");`